src/libraries/Common/tests/System/Net/Http/Http3LoopbackStream.cs

```diff
@@ -199,6 +211,17 @@ static int EncodeHttpInteger(long longToEncode, Span<byte> buffer)
             }
         }
 
+        public async Task<int> ReadDataStreamAsync(byte[] payload)
+        {
+            int bytesRead = await _stream.ReadAsync(payload.AsMemory(0)).ConfigureAwait(false);
+            return bytesRead;
+        }
+
+        public async Task SendDataStreamAsync(byte[] payload)
+        {
+            await _stream.WriteAsync(payload.AsMemory(), true);
```

Comment:
Yes it is `completeWrites`
I wirte on the same stream multiple times but from different directions, but it also works writing from the same direction multiple times, I will add a multiple send test tomorrow after I change the streams to quic

Reply:
Different directions are different things. If you pass `completeWrites` you are not able to send data in this direction anymore.
That tells me that there were no tests that would send data from server to client multiple times on the same stream, otherwise this test will not work.